sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -3842,9 +3842,16 @@ pi_result piProgramBuild(pi_program Program, pi_uint32 NumDevices,
   if (ZeResult != ZE_RESULT_SUCCESS) {
     // We adjust pi_program below to avoid attempting to release zeModule when
     // RT calls piProgramRelease().
-    ZeModule = nullptr;
     Program->State = _pi_program::Invalid;
     Result = mapError(ZeResult);
+    if (Program->ZeBuildLog) {
+      ZE_CALL_NOCHECK(zeModuleBuildLogDestroy, (Program->ZeBuildLog));
```

Comment:
@bso-intel, @againull : this change causes a regression that build-log is destroyed before SYCL RT reads it (to report to the user). Please revert (or re-work) the patch